src/irmin-pack/unix/snapshot.ml

```diff
@@ -104,8 +104,11 @@ module Make (Args : Args) = struct
         io_read_and_decode_entry_prefix ~off:offset t
       in
       let entry_of_hash hash = key_of_hash hash t.inode_pack in
+      (* Bytes.unsafe_to_string usage: buf is created locally, uniquely owned; we assume
+         Dispatcher.read_exn returns unique ownership; then call to Bytes.unsafe_to_string
```

Comment:
For the "assume `Module.func` returns unique ownership" cases, I'm thinking that we should also document this behavior in the documentation for the function so that it is more likely to be seen by future modifiers of the code. In general, I think any API we write that accepts a buffer as input ought to never take ownership of it beyond the lifecycle of the function call. My proposal here is that a buffer as input is a signal to indicate this kind of pattern.

Reply:
Yes, this would be good to do, I agree. But I don't want that extra work to block this PR getting merged (this PR fixes bugs that are in the GC code, so probably somewhat pressing). I suggest to open another issue. Hope this makes sense.

Reply:
Let's open another issue to document those. :+1:

Reply:
Oh thank you Kevin! You are too kind. You are right that those functions should be documented, but yes, let's merge this and open a follow-up issue. Thank you!!!